### PR TITLE
Data insight workflow

### PR DESCRIPTION
**Pull request by TeddyCr**
### Describe your changes :

I worked on the ..... because ...

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

**Review by ayush-shah — DISMISSED** [review]

File: openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/data_insight.py [hunk 1]
```diff
@@ -37,9 +47,47 @@ def build_data_insight_workflow_config(
     Given an airflow_pipeline, prepare the workflow config JSON
     """
 
+    try:
+        metadata = OpenMetadata(config=ingestion_pipeline.openMetadataServerConnection)
+    except Exception as exc:
+        raise ClientInitializationError(f"Failed to initialize the client: {exc}")
+
+    openmetadata_service = metadata.get_by_name(
+        entity=MetadataService, fqn=OPENMETADATA_SERVICE_FQN
```
Comment (pmbrull) [inline openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/data_insight.py:56]: Hi @TeddyCr actually I moved this out. Might be cleaner to pick up the name from the `ingestion_pipeline.service.fullyQualifiedName`

File: ingestion/src/metadata/utils/constants.py [hunk 2]
```diff
@@ -16,3 +16,14 @@
 DOT = "_DOT_"
 TEN_MIN = 10 * 60
 UTF_8 = "utf-8"
+
+OPENMETADATA_SERVICE_FQN = "Openmetadata"
```
Comment (pmbrull) [inline ingestion/src/metadata/utils/constants.py:20]:
```suggestion
```

File: openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/data_insight.py [hunk 3]
```diff
@@ -12,22 +12,32 @@
...
     Processor,
-    WorkflowConfig,
+    Sink,
+)
+from metadata.generated.schema.metadataIngestion.workflow import (
+    Source as WorkflowSource,
 )
+from metadata.generated.schema.metadataIngestion.workflow import WorkflowConfig
+from metadata.ingestion.ometa.ometa_api import OpenMetadata
+from metadata.utils.constants import ES_SOURCE_TO_ES_OBJ_ARGS, OPENMETADATA_SERVICE_FQN
```
Comment (pmbrull) [inline openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/data_insight.py:40]:
```suggestion
from metadata.utils.constants import ES_SOURCE_TO_ES_OBJ_ARGS
```

File: openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/data_insight.py (hunk 1 above)
Comment (TeddyCr) [reply openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/data_insight.py:56]: I'll update it

Comment (pmbrull) [inline openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/data_insight.py:56]:
```suggestion
        entity=MetadataService, fqn=ingestion_pipeline.service.fullyQualifiedName
```

**Review by pmbrull — DISMISSED** [review]

**Review by pmbrull — DISMISSED** [review]

**Review by harshach — APPROVED** [review]

**Pull request merged**